### ISSUE TITLE
data binding with relationships seems broken >= 3.0.0

### ISSUE THREAD

**Issue by st-h** [emberjs/data#5444]
following workflow:
- create a record
- update any property of the record
- save the record
- observe the updated property via its parent in a template:
  - {{#each parent.entities as |entity|}} {{entity.prop}} {{/each}}

==> the property is never updated in the template. However Inspector shows correct values.
- if the created record is stored in a property via `this.set('_entity', this.get('store').createRecord('entity', {  parent: parent' }));` and `_entity.prop` is rendered, this  just works fine.

I created a simple app to reproduce this behaviour: https://github.com/st-h/ember-data-binding

#5389 might be related

**Comment by st-h** [comment]
seems to be fixed in `3.2.0-beta.2`. Will keep the reproducer online until the release is out in case it is needed.

**Comment by runspired** [comment]
@st-h I'm having a little trouble understanding in your reproduction app what is "necessary to reproduce" vs what merely part of your setup.

- is using the local-storage addon a requirement?
- is the parent record also being `new` a requirement?

**Comment by runspired** [comment]
After debugging I can confirm this is a bug we fixed caused by changes in Ember. The initial fix should have been back-ported but wasn't, and is also not easy to backport (the initial fix was problematic and follow up changes were made). You can either use an earlier version of Ember, use ember-data beta, or change the createRecord code to the following until this is backported or a stable release with the fix is available.

Closing as this is not a new issue.

**Comment by runspired** [comment]
cc @bmac as it was already on our radar to backport the properties-in-init fixes to 2.18. I can help get some PRs up later, there are multiple commits involved.

**Comment by st-h** [comment]
sorry. didn't mean to cause any trouble by leaving this issue open. Thought it was clear that I am no longer seeing this in beta. Happy to see that this is already fixed.

**Comment by runspired** [comment]
@st-h oh you did nothing bad :)

I leave issues open until I've confirmed they are no-longer an issue and had a chance to do a root-cause-analysis.  While you were correct that it was fixed on beta, I needed to ensure it was a known-fix vs a "mysteriously works again" fix, in which case additional fixes to backport would have been needed and new test coverage added.

Fwiw, this is now under extensive test coverage :)

https://github.com/emberjs/data/pull/5413

Thanks again for opening the issue and the simple reproduction!  I'm especially happy that you were able to realize that this was what folks were talking about in #5389
